db_go18_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2017 Square Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build go1.8
+
+package squalor
+
+import (
+	"context"
+	"testing"
+	"time"
+)
+
+func TestWithContextTimeout(t *testing.T) {
+	db := makeTestDB(t, usersDDL)
+	defer db.Close()
+
+	if db.GetContext() == nil {
+		t.Fatal("Expected default non-nil Context on DB")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
```

Comment:
[`context.WithTimeout()`](https://github.com/golang/go/blob/cd6b6202dd1559b3ac63179b45f1833fcfbe7eca/src/context/context.go#L436) tests basically the same thing as `context.WithDeadline()`. You don't need both tests here.